### PR TITLE
improvements fall2017

### PR DESCRIPTION
**Pull request by BenGardiner**

### CODE REVIEW

**Comment by BenGardiner** [comment]
the most recent commit resolves #6

File: scripts/generate-descriptions-md-template [hunk 1]
```diff
@@ -25,20 +25,18 @@ tops = list()
...
+		set_collapsed(node)
 
 	for key, value in node.get('ideas', dict()).iteritems():
 		truncate(value, cuts, False)
 
 	return node
 
 def get_safe_filename(notsafe):
-	safe = notsafe.replace("?", "_").replace("/", "_").replace(",", "_").replace('\'','_').replace('(','_').replace(')','_').replace('  ', ' ')
+	safe = notsafe.replace("?", "_").replace("/", "_").replace(",", "_").replace('\'','_').replace('"','_').replace('(','_').replace(')','_').replace('  ', ' ')
```
Comment (JonathanBeverley) [inline scripts/generate-descriptions-md-template]: This is getting unwieldy. Maybe prefer something like:
``safe = re.sub("[?/,'\"()\t]", '_', notsafe)``

**Comment by JonathanBeverley** [comment]
What is the rendercache script for?

**Comment by BenGardiner** [comment]
> What is the rendercache script for?

the rendermup script is pretty slow right now. And it is run at least once for each subtree rendered in the report. The rendercache script wraps rendermup in a hash-based cache. Speeds up the iterative report generation.

**Comment by BenGardiner** [comment]
this is all in #18 now

**Pull request closed without merging**